### PR TITLE
Remove dependency on Quick/Nimble

### PR DESCRIPTION
**Pull request by yoichitgy**
This PR addresses the same issue of [Swinject Issue #472](https://github.com/Swinject/Swinject/issues/472), and performs the migration from Quick/Nimble to XCTest in the same way as [Swinject PR #473](https://github.com/Swinject/Swinject/pull/473).

- Replaced `XxxSpec` using Quick/Nimble with `XxxTests` using XCTest.
- Removed dependency on Quick/Nimble. (We still need Carthage because of the dependency on Swinject.)

**Note**
A few unit tests that check throwing `precondition` were commented out because those cannot be tested without depending on Nimble or CwlPreconditionTesting. I prioritized migration to XCTest and reducing dependency this time for easier maintainability as stated in #472.

### CODE REVIEW

File: Tests/SwinjectAutoregistrationTests/AutoregistrationTests.swift [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+        container.autoregister(UnwrappedService.self, initializer: UnwrappedService.init)
+        let service = container.resolve(UnwrappedService.self)
+        XCTAssertNotNil(service)
+    }
+
+    #if !SWIFT_PACKAGE && arch(x86_64)
+    func testAssertionIsThrownWhenSameTypeArgumentsArePassed() {
+        // For now we just comment out the test for `precondition()` because we cannot do it without depending on Nimble or CwlPreconditionTesting.
+        
+        // container.autoregister(SameArgumentsService.self, arguments: String.self, Int.self, String.self, initializer: SameArgumentsService.init)
```
Comment (yoichitgy) [inline Tests/SwinjectAutoregistrationTests/AutoregistrationTests.swift:215]: Commented out `precondition` test here.

File: Tests/SwinjectAutoregistrationTests/AutoregistrationTests.swift [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    }
+
+    func testAssertionIsThrownWhenTryingToResolveServiceWithTooManyDependencies() {
+        // For now we just comment out the test for `precondition()` because we cannot do it without depending on Nimble or CwlPreconditionTesting.
+
+        // container.autoregister(Service21.self, initializer: Service21.init)
+        // _ = container.resolve(Service21.self)
```
Comment (yoichitgy) [inline Tests/SwinjectAutoregistrationTests/AutoregistrationTests.swift:222]: Commented out `precondition` test here.

File: Tests/SwinjectAutoregistrationTests/OperatorsTests.swift [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        // For now we just comment out the test for `precondition()` because we cannot do it without depending on Nimble or CwlPreconditionTesting.
+
+        // var logs = [String]()
+        // Container.loggingFunction = { logs.append($0) }
+        //
+        // let expectation: Expectation<Void> = expect {
+        //    _ = container ~> (Service0.self, name: "NonExistentService")
+        // }
+        //
+        // XCTAssertEqual(logs.count, 1)
```
Comment (yoichitgy) [inline Tests/SwinjectAutoregistrationTests/OperatorsTests.swift:69]: Commented out `precondition` test here.

File: Tests/SwinjectAutoregistrationTests/AutoregistrationTests.swift (hunk 1 above)
Comment (mpdifran) [reply Tests/SwinjectAutoregistrationTests/AutoregistrationTests.swift:215]: Should we think about maybe making this a `throw`ing function that throws a custom error when the arguments are not unique types? That would make it easier to test. We could also mock `precondition()` using an injected closure, but I think making this `throw`ing is the best way.

**Review by mpdifran — APPROVED** [review]

File: Tests/SwinjectAutoregistrationTests/AutoregistrationTests.swift (hunk 1 above)
Comment (yoichitgy) [reply Tests/SwinjectAutoregistrationTests/AutoregistrationTests.swift:215]: `throw`ing an error is a solution if we accept a breaking change (The users of the API must add `try` where `autoregister` is used). If we take the approach, SwinjectAutoregistration version will go to v3.

Another approach is adding a dependency on `CwlPreconditionTesting`. The project is pretty stable where the last release was on Oct 24, 2019 but it keeps working used by the latest versions of Quick/Nimble. The dependency on `CwlPreconditionTesting` looks not so bad.

**Comment by yoichitgy** [comment]
@mpdifran Thank you for the review😃

**Pull request merged**